controllers/iamrole_controller.go

```diff
@@ -140,16 +141,30 @@ func (r *IamroleReconciler) HandleReconcile(ctx context.Context, req ctrl.Reques
 		ns = *ns2
 	}
 
+	// Figure out the name of the IAM Role itself that will be created based on the Iamrole resource name, namespace,
+	// and the configuration settings.
 	roleName, err := utils.GenerateRoleName(ctx, iamRole, *config.Props, &ns)
-	log.V(1).Info("roleName constructed successfully", "roleName", roleName)
-
 	if err != nil {
 		r.Recorder.Event(iamRole, v1.EventTypeWarning, string(iammanagerv1alpha1.Error), "Unable to construct iam role name to error "+err.Error())
 		// It is not clear to me that we want to requeue here - as this is a fairly permanent
 		// error. Is there a better pattern here?
 		return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
 	}
 
+	// Determine whether or not this Iamrole is supposed to be connected to a ServiceAccount or not. This is
+	// used during the 'Ready' and 'default' status case handlers below to ensure the connection between the
+	// ServiceAccount and IAM Role (if desired).
+	shouldManageSa, saName := utils.ParseIRSAAnnotation(ctx, iamRole)
```

Comment:
maybe move this down to right above where you check `shouldManageSa` instead of having it up here

Reply:
I could move this down - but then it has to be duplicated (once inside the `iammanagerv1alpha1.Ready` case and once in the `default` case. Do you think that's more clear though and worth the duplication?

Reply:
I probably would, just to reduce the distance I have to scan to remember how this got set, but that's more of a style thing. I  just really don't like these boolean "do something later" style flags, and the less I have to scroll back to figure out how it got set, the better.